### DISCUSSION TITLE
One click/Quick add personalised columns to both left AND right of list

### DISCUSSION THREAD

**Discussion by CritchCode** [arnoudkooi/SN-Utils#364] [Ideas]
Would you consider adding an option to personalise 'one-click/quick add' fields at the left and right of any list?
The default could be blank for the left column but in my case I'm regularly checking sys_ids on many tables, particularly when doing imports, and this would be very helpful. (Mock-up below)
![SNUtilsIdea-LeftRightOneClick](https://user-images.githubusercontent.com/55980966/218678786-feb49996-2c77-4ab4-821e-0e75a8df87c7.png)

![image](https://user-images.githubusercontent.com/55980966/218679039-3a2ea81b-0dda-4720-9505-5221b05af7b7.png)

**Comment by CritchCode** [comment]
I should add that currently, I manually go into the personalise list dialog, use SN Utils awesome double-click feature to show the 'add dotwalk field' box, then add sys_id, then move it to the top. It gets a bit repetitive when doing the same in multiple tables on multiple instances. Thanks!

**Comment by arnoudkooi** [comment]
The extra setting would be confusing for most users, and I see the only use-case for the sys_id
I have added it in 2 ways.

Via the special button, when holding shift while clicking, it will add the sys_id to the beginning of the list.

![Google Chrome20230214 - 2041](https://user-images.githubusercontent.com/840781/218841594-3ce75f58-4129-4f45-8206-e3f2cd4b2ecb.png)

As an alternative (and if the SN Utils button is not visible), you can trigger the Technical Names via a double-click and then click the +sys_id link
![Google Chrome20230214 - 2047](https://user-images.githubusercontent.com/840781/218844723-3b12f0c0-ec0a-41ab-abff-96b768738a34.png)

**Reply by CritchCode** [reply]
That's fantastic! This is simply the best browser extension I've ever installed. Thank you so much Arnoud

**Reply by CritchCode** [reply]
Which version will this be included in, so I can watch out for it? Thanks

**Reply by arnoudkooi** [reply]
6.3.6.0 The Safari version should be available

**Reply by CritchCode** [reply]
I forgot to come back and say thank you! This one new feature has saved me tons of time already. Holding shift while clicking the button is definitely my preferred method as it's just so fast. It really helps when doing a ton of data reconciliation or tracing through debugging things and checking out records by sys_id. Much appreciated! @arnoudkooi

**Reply by CritchCode** [reply]
... and yes, I'm both a MacOS Safari user (preference!) and a Windows + Chrome/Firefox user for some projects where I'm compelled to use provided equipment.

**Comment by codaroma** [comment]
I have my "One click personalise list fields:" set to
sys_scope,sys_created_on,sys_created_by,sys_updated_on,sys_updated_by,sys_id
But the sys_id is not being added to the end of the list columns.
Is this a bug?

**Reply by arnoudkooi** [reply]
It can only add columns that are on the left side of the list collector but I could add an exception for sys_id
Are the 2 options above not better, rather than adding sys_id by default? What is your reason for wanting it?

**Reply by codaroma** [reply]
Thanks, no worries. I will use the 2 existing options.
